tools/lsp/common/rename_component.rs

```diff
@@ -0,0 +1,781 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.2 OR LicenseRef-Slint-commercial
+
+use std::path::Path;
+
+use crate::{common, util};
+
+use i_slint_compiler::{
+    diagnostics::{SourceFile, Spanned},
+    object_tree,
+    parser::{syntax_nodes, SyntaxKind},
+    typeloader::TypeLoader,
+};
+
+fn symbol_export_names(document_node: &syntax_nodes::Document, type_name: &str) -> Vec<String> {
+    let mut result = vec![];
+
+    for export in document_node.ExportsList() {
+        for specifier in export.ExportSpecifier() {
+            if specifier.ExportIdentifier().text() == type_name {
+                result.push(
+                    specifier
+                        .ExportName()
+                        .map(|n| n.text().to_string())
+                        .unwrap_or_else(|| type_name.to_string()),
+                );
+            }
+        }
+        if let Some(component) = export.Component() {
+            if component.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+        for structs in export.StructDeclaration() {
+            if structs.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+        for enums in export.EnumDeclaration() {
+            if enums.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+    }
+
+    result.sort();
+    result
+}
+
+fn replace_element_types(
+    element: &syntax_nodes::Element,
+    old_type: &str,
+    new_type: &str,
+    edits: &mut Vec<(SourceFile, lsp_types::TextEdit)>,
+) {
+    if let Some(name) = element.QualifiedName() {
+        if name.text().to_string().trim() == old_type {
+            edits.push((
+                element.source_file.clone(),
+                lsp_types::TextEdit {
+                    range: util::map_node(&name).expect("Just had a source_file"),
+                    new_text: new_type.to_string(),
+                },
+            ))
+        }
+    }
+
+    for c in element.children() {
+        match c.kind() {
+            SyntaxKind::SubElement => {
+                let e: syntax_nodes::SubElement = c.into();
+                replace_element_types(&e.Element(), old_type, new_type, edits);
+            }
+            SyntaxKind::RepeatedElement => {
+                let e: syntax_nodes::RepeatedElement = c.into();
+                replace_element_types(&e.SubElement().Element(), old_type, new_type, edits);
+            }
+            SyntaxKind::ConditionalElement => {
+                let e: syntax_nodes::ConditionalElement = c.into();
+                replace_element_types(&e.SubElement().Element(), old_type, new_type, edits);
+            }
+            _ => { /* do nothing */ }
```

Comment:
```suggestion
            _ => {}
```

Reply:
I wonder if it would be not wiser and easier to go over all tokens that matches the name and check if it is actually that component.

Reply:
`_ => {}` is a known pattern in rust.
To me, this kind of comments are just like `i += 1; // add one to i`
Unless you want to provide an extra reason why we shouldn't do anything in there